Add tasks to fix RHEL support

diademiemi opened the pull request:
## Pull request type
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?
RHEL 7 and RHEL 8 have different repositories and require other tasks to get Checkmk to install.  
CentOS 7 does not require any repositories, but RHEL 7 does.  
RHEL 8 also requires different repositories than CentOS 8, namely CRB and EPEL instead of powertools.  

## What is the new behavior?
* CRB is enabled on RHEL 8.  
* EPEL is enabled on RHEL 8.
  This requires a package URL, as epel-release is not a package in RHEL. https://www.redhat.com/en/blog/whats-epel-and-how-do-i-use-it  
* Powertools is enabled on CentOS 8.  
* Powertools (Extra repos) is enabled on CentOS 7.  

## Other information
I've tested the changes on CentOS 7, CentOS 8 Stream, RHEL 7.9 and RHEL 8.6.  
If you want to test these changes in Vagrant, I wrote a little guide to get RHEL in Vagrant: https://gist.github.com/diademiemi/799949a7a3116eb301c6cbc3b9884ca5  

Checkmk does not support RHEL 9 and it's derivatives yet, but once it does it should be largely the same as CentOS & RHEL 8.

diademiemi commented on roles/server/tasks/sites.yml:
```diff
@@ -21,6 +21,7 @@
     omd version {{ item.name }} | egrep -o '[^ ]+$'
   args:
     executable: /bin/bash
+  no_log: true
```
Quick fix to prevent showing admin passwd in log

robin-tribe29 replied:
This should have been there from the beginning, did we miss it?

diademiemi commented on roles/server/tasks/RedHat.yml:
```diff
@@ -26,6 +85,7 @@
     - set-selinux-boolean
 
 - name: "Make sure firewalld is started and enabled"
+  become: true
```
Fix required become

diademiemi commented:
Integration test fail seems unrelated. Also happens in the run for the 0.9.0 PR: https://github.com/tribe29/ansible-collection-tribe29.checkmk/runs/8251666358?check_suite_focus=true

mto79 commented:
Very Nice @diademiemi ! Thanks!

robin-tribe29 reviewed (commented):
I get the following error on CentOS Stream 8:
`Depsolve Error occurred: \n Problem: conflicting requests\n  - nothing provides libdbi needed by check-mk-enterprise-2.1.0p11-el8-38.x86_64`
Do we need yet different handling on CentOS Stream?

robin-tribe29 commented:
And yes, integration tests are broken, and I am working to replace them with molecule (#145), but still working on the finishing touches there.

diademiemi commented on roles/server/tasks/RedHat.yml:
```diff
@@ -1,10 +1,70 @@
+  - name: "Download EPEL GPG Key on RHEL 8."
+    ansible.builtin.get_url:
+      url: "{{ checkmk_server_epel_gpg_key }}"
+      dest: "/tmp/EPEL-pubkey.gpg"
```
Sure will do

diademiemi commented on roles/server/tasks/RedHat.yml:
```diff
@@ -1,10 +1,70 @@
+
+  - name: "Enable CodeReady Linux Builder repo on RHEL 8."
+    become: true
+    community.general.rhsm_repository:
```
`community.general` should be installed by default on most installations of Ansible. We already use modules that aren't in `ansible.builtin` elsewhere (like `ansible.posix`) so only having `ansible-core` wouldn't be anough to use this collection regardless.

robin-tribe29 replied:
I added the dependencies, will do some local testing and push the change later. For this PR it is fine, however.

diademiemi commented:
> I get the following error on CentOS Stream 8:
> `Depsolve Error occurred: \n Problem: conflicting requests\n - nothing provides libdbi needed by check-mk-enterprise-2.1.0p11-el8-38.x86_64`
> Do we need yet different handling on CentOS Stream?

Hmm. I did not get this error. I'm using the `generic/centos8s` Vagrant box to test and this role runs for me

robin-tribe29 commented:
> > I get the following error on CentOS Stream 8:
> > `Depsolve Error occurred: \n Problem: conflicting requests\n - nothing provides libdbi needed by check-mk-enterprise-2.1.0p11-el8-38.x86_64`
> > Do we need yet different handling on CentOS Stream?
> 
> Hmm. I did not get this error. I'm using the `generic/centos8s` Vagrant box to test and this role runs for me

I am using `centos/stream8`. But it is not a blocker for me. If someone experiences the same issue, we can always investigate then.

diademiemi commented:
`server1`: RHEL 8  
`server2`: RHEL 7  
`server3`: CentOS 7  
`server4`: CentOS 8 Stream  

My output
<details> <summary> ansible-playbook server.yml -i inventory.ini </summary>

```
PLAY [server] *******************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] **********************************************************************************************************************************************************************************************************************
ok: [server3]
ok: [server2]
ok: [server4]
ok: [server1]

TASK [tribe29.checkmk.server : Preflight - Fail if target host is not stable for this Role.] ************************************************************************************************************************************************
skipping: [server1]
skipping: [server2]
skipping: [server3]
skipping: [server4]

TASK [tribe29.checkmk.server : Include OS Family specific Variables.] ***********************************************************************************************************************************************************************
ok: [server1]
ok: [server2]
ok: [server3]
ok: [server4]

TASK [tribe29.checkmk.server : Install Checkmk Prerequisites.] ******************************************************************************************************************************************************************************
ok: [server3] => (item=['firewalld', 'dnf-plugins-core', 'libsemanage-python'])
ok: [server2] => (item=['firewalld', 'libsemanage-python'])
ok: [server1] => (item=['firewalld', 'dnf-plugins-core', 'python3-libsemanage'])
ok: [server4] => (item=['firewalld', 'dnf-plugins-core', 'python3-libsemanage'])

TASK [tribe29.checkmk.server : Download Checkmk Server Setup.] ******************************************************************************************************************************************************************************
changed: [server4]
changed: [server2]
changed: [server1]
changed: [server3]

TASK [tribe29.checkmk.server : Download Checkmk GPG Key.] ***********************************************************************************************************************************************************************************
changed: [server3]
changed: [server4]
changed: [server1]
changed: [server2]

TASK [tribe29.checkmk.server : Import Checkmk GPG Key.] *************************************************************************************************************************************************************************************
skipping: [server1]
skipping: [server2]
skipping: [server3]
skipping: [server4]

TASK [tribe29.checkmk.server : Verify Checkmk Setup.] ***************************************************************************************************************************************************************************************
skipping: [server1]
skipping: [server2]
skipping: [server3]
skipping: [server4]

TASK [tribe29.checkmk.server : Print Verification Output.] **********************************************************************************************************************************************************************************
skipping: [server1]
skipping: [server2]
skipping: [server3]
skipping: [server4]

TASK [tribe29.checkmk.server : Import Checkmk GPG Key.] *************************************************************************************************************************************************************************************
changed: [server3]
changed: [server2]
changed: [server4]
changed: [server1]

TASK [tribe29.checkmk.server : Include OS Family specific Playbook.] ************************************************************************************************************************************************************************
included: /home/diademiemi/Desktop/Work/RHTest/ansible_collections/tribe29/checkmk/roles/server/tasks/RedHat.yml for server1, server2, server3, server4

TASK [tribe29.checkmk.server : Enable CodeReady Linux Builder repo on RHEL 8.] **************************************************************************************************************************************************************
skipping: [server2]
skipping: [server3]
skipping: [server4]
ok: [server1]

TASK [tribe29.checkmk.server : Download EPEL GPG Key on RHEL 8.] ****************************************************************************************************************************************************************************
skipping: [server2]
skipping: [server3]
skipping: [server4]
changed: [server1]

TASK [tribe29.checkmk.server : Import Checkmk GPG Key on RHEL 8.] ***************************************************************************************************************************************************************************
skipping: [server2]
skipping: [server3]
skipping: [server4]
changed: [server1]

TASK [tribe29.checkmk.server : Install epel-release from URL on RHEL 8.] ********************************************************************************************************************************************************************
skipping: [server2]
skipping: [server3]
skipping: [server4]
changed: [server1]

TASK [tribe29.checkmk.server : Enable powertools repository on CentOS 8.] *******************************************************************************************************************************************************************
skipping: [server1]
skipping: [server2]
skipping: [server3]
changed: [server4]

TASK [tribe29.checkmk.server : Enable powertools repository on RHEL 7.] *********************************************************************************************************************************************************************
skipping: [server1] => (item=rhel-7-server-optional-rpms) 
skipping: [server1] => (item=rhel-7-server-extras-rpms) 
skipping: [server3] => (item=rhel-7-server-optional-rpms) 
skipping: [server3] => (item=rhel-7-server-extras-rpms) 
skipping: [server4] => (item=rhel-7-server-optional-rpms) 
skipping: [server4] => (item=rhel-7-server-extras-rpms) 
ok: [server2] => (item=rhel-7-server-optional-rpms)
ok: [server2] => (item=rhel-7-server-extras-rpms)

TASK [tribe29.checkmk.server : Install Checkmk Server.] *************************************************************************************************************************************************************************************
changed: [server4]
changed: [server3]
changed: [server1]
changed: [server2]

TASK [tribe29.checkmk.server : Enable httpd can network connect selinux boolean.] ***********************************************************************************************************************************************************
changed: [server3]
changed: [server2]
changed: [server1]
changed: [server4]

TASK [tribe29.checkmk.server : Make sure firewalld is started and enabled] ******************************************************************************************************************************************************************
ok: [server3]
ok: [server2]
ok: [server4]
ok: [server1]

TASK [tribe29.checkmk.server : Open firewall ports.] ****************************************************************************************************************************************************************************************
changed: [server3] => (item=80/tcp)
changed: [server4] => (item=80/tcp)
changed: [server2] => (item=80/tcp)
changed: [server1] => (item=80/tcp)
changed: [server3] => (item=443/tcp)
changed: [server4] => (item=443/tcp)
changed: [server2] => (item=443/tcp)
changed: [server1] => (item=443/tcp)
changed: [server3] => (item=8000/tcp)
changed: [server4] => (item=8000/tcp)
changed: [server2] => (item=8000/tcp)
changed: [server1] => (item=8000/tcp)

TASK [tribe29.checkmk.server : Include Site Management Playbook.] ***************************************************************************************************************************************************************************
included: /home/diademiemi/Desktop/Work/RHTest/ansible_collections/tribe29/checkmk/roles/server/tasks/sites.yml for server1, server2, server3, server4

TASK [tribe29.checkmk.server : Create Sites.] ***********************************************************************************************************************************************************************************************
changed: [server3] => (item=None)
changed: [server3]
changed: [server2] => (item=None)
changed: [server2]
changed: [server4] => (item=None)
changed: [server4]
changed: [server1] => (item=None)
changed: [server1]

TASK [tribe29.checkmk.server : Get site version.] *******************************************************************************************************************************************************************************************
ok: [server4] => (item=None)
ok: [server4]
ok: [server3] => (item=None)
ok: [server3]
ok: [server2] => (item=None)
ok: [server2]
ok: [server1] => (item=None)
ok: [server1]

TASK [tribe29.checkmk.server : Include update site tasks.] **********************************************************************************************************************************************************************************
skipping: [server1] => (item={'changed': False, 'stdout': '2.1.0p11.cme', 'stderr': '', 'rc': 0, 'cmd': "set -o pipefail\nomd version site | egrep -o '[^ ]+$'\n", 'start': '2022-09-12 05:42:58.209282', 'end': '2022-09-12 05:42:58.582975', 'delta': '0:00:00.373693', 'msg': '', 'invocation': {'module_args': {'executable': '/bin/bash', '_raw_params': "set -o pipefail\nomd version site | egrep -o '[^ ]+$'\n", '_uses_shell': True, 'warn': False, 'stdin_add_newline': True, 'strip_empty_ends': True, 'argv': None, 'chdir': None, 'creates': None, 'removes': None, 'stdin': None}}, 'stdout_lines': ['2.1.0p11.cme'], 'stderr_lines': [], 'failed': False, 'item': {'name': 'site', 'version': '2.1.0p11', 'state': 'started', 'admin_pw': 'testing', 'update_conflict_resolution': 'install'}, 'ansible_loop_var': 'item'}) 
skipping: [server2] => (item={'changed': False, 'end': '2022-09-12 05:42:58.372740', 'stdout': '2.1.0p11.cme', 'cmd': "set -o pipefail\nomd version site | egrep -o '[^ ]+$'\n", 'start': '2022-09-12 05:42:58.238200', 'delta': '0:00:00.134540', 'stderr': '', 'rc': 0, 'invocation': {'module_args': {'creates': None, 'executable': '/bin/bash', '_uses_shell': True, 'strip_empty_ends': True, '_raw_params': "set -o pipefail\nomd version site | egrep -o '[^ ]+$'\n", 'removes': None, 'argv': None, 'warn': False, 'chdir': None, 'stdin_add_newline': True, 'stdin': None}}, 'msg': '', 'stdout_lines': ['2.1.0p11.cme'], 'stderr_lines': [], 'failed': False, 'item': {'name': 'site', 'version': '2.1.0p11', 'state': 'started', 'admin_pw': 'testing', 'update_conflict_resolution': 'install'}, 'ansible_loop_var': 'item'}) 
skipping: [server3] => (item={'changed': False, 'end': '2022-09-12 09:42:58.324017', 'stdout': '2.1.0p11.cme', 'cmd': "set -o pipefail\nomd version site | egrep -o '[^ ]+$'\n", 'start': '2022-09-12 09:42:56.459174', 'delta': '0:00:01.864843', 'stderr': '', 'rc': 0, 'invocation': {'module_args': {'creates': None, 'executable': '/bin/bash', '_uses_shell': True, 'strip_empty_ends': True, '_raw_params': "set -o pipefail\nomd version site | egrep -o '[^ ]+$'\n", 'removes': None, 'argv': None, 'warn': False, 'chdir': None, 'stdin_add_newline': True, 'stdin': None}}, 'msg': '', 'stdout_lines': ['2.1.0p11.cme'], 'stderr_lines': [], 'failed': False, 'item': {'name': 'site', 'version': '2.1.0p11', 'state': 'started', 'admin_pw': 'testing', 'update_conflict_resolution': 'install'}, 'ansible_loop_var': 'item'}) 
skipping: [server4] => (item={'changed': False, 'stdout': '2.1.0p11.cme', 'stderr': '', 'rc': 0, 'cmd': "set -o pipefail\nomd version site | egrep -o '[^ ]+$'\n", 'start': '2022-09-12 09:42:58.135817', 'end': '2022-09-12 09:42:58.244554', 'delta': '0:00:00.108737', 'msg': '', 'invocation': {'module_args': {'executable': '/bin/bash', '_raw_params': "set -o pipefail\nomd version site | egrep -o '[^ ]+$'\n", '_uses_shell': True, 'warn': False, 'stdin_add_newline': True, 'strip_empty_ends': True, 'argv': None, 'chdir': None, 'creates': None, 'removes': None, 'stdin': None}}, 'stdout_lines': ['2.1.0p11.cme'], 'stderr_lines': [], 'failed': False, 'item': {'name': 'site', 'version': '2.1.0p11', 'state': 'started', 'admin_pw': 'testing', 'update_conflict_resolution': 'install'}, 'ansible_loop_var': 'item'}) 

TASK [tribe29.checkmk.server : Start Sites.] ************************************************************************************************************************************************************************************************
changed: [server4] => (item=None)
changed: [server4]
changed: [server3] => (item=None)
changed: [server3]
changed: [server2] => (item=None)
changed: [server2]
changed: [server1] => (item=None)
changed: [server1]

TASK [tribe29.checkmk.server : Stop Sites.] *************************************************************************************************************************************************************************************************
skipping: [server1] => (item=None) 
skipping: [server1]
skipping: [server2] => (item=None) 
skipping: [server2]
skipping: [server3] => (item=None) 
skipping: [server3]
skipping: [server4] => (item=None) 
skipping: [server4]

TASK [tribe29.checkmk.server : Destroy Sites.] **********************************************************************************************************************************************************************************************
skipping: [server1] => (item=None) 
skipping: [server2] => (item=None) 
skipping: [server1]
skipping: [server3] => (item=None) 
skipping: [server2]
skipping: [server3]
skipping: [server4] => (item=None) 
skipping: [server4]

TASK [tribe29.checkmk.server : Update Site Admin Password.] *********************************************************************************************************************************************************************************
changed: [server2] => (item=None)
changed: [server2]
changed: [server3] => (item=None)
changed: [server3]
changed: [server4] => (item=None)
changed: [server4]
changed: [server1] => (item=None)
changed: [server1]
```

</details>

I can see they're all running after that

diademiemi commented:
> > > I get the following error on CentOS Stream 8:
> > > `Depsolve Error occurred: \n Problem: conflicting requests\n - nothing provides libdbi needed by check-mk-enterprise-2.1.0p11-el8-38.x86_64`
> > > Do we need yet different handling on CentOS Stream?
> > 
> > 
> > Hmm. I did not get this error. I'm using the `generic/centos8s` Vagrant box to test and this role runs for me
> 
> I am using `centos/stream8`. But it is not a blocker for me. If someone experiences the same issue, we can always investigate then.

Ah okay that probably explains it. That image hasn't had a release in over a year, the `generic/centos8s` image is still maintained and had a release 13 days ago.
